src/main/java/net/fabricmc/loom/util/ZipReprocessorUtil.java

```diff
@@ -95,44 +95,80 @@ public static void reprocessZip(File file, boolean reproducibleFileOrder, boolea
 			return;
 		}
 
-		try (ZipFile zipFile = new ZipFile(file)) {
+		try (var zipFile = new ZipFile(file)) {
 			ZipEntry[] entries;
 
 			if (reproducibleFileOrder) {
-				entries = zipFile.stream().sorted(Comparator.comparing(ZipEntry::getName, ZipReprocessorUtil::specialOrdering)).toArray(ZipEntry[]::new);
+				entries = zipFile.stream()
+						.sorted(Comparator.comparing(ZipEntry::getName, ZipReprocessorUtil::specialOrdering))
+						.toArray(ZipEntry[]::new);
 			} else {
-				entries = zipFile.stream().toArray(ZipEntry[]::new);
+				entries = zipFile.stream()
+						.toArray(ZipEntry[]::new);
 			}
 
-			ByteArrayOutputStream outZip = new ByteArrayOutputStream(zipFile.size());
+			final var outZip = new ByteArrayOutputStream(entries.length);
 
-			try (ZipOutputStream zipOutputStream = new ZipOutputStream(outZip)) {
+			try (var zipOutputStream = new ZipOutputStream(outZip)) {
 				for (ZipEntry entry : entries) {
 					ZipEntry newEntry = entry;
 
 					if (!preserveFileTimestamps) {
 						newEntry = new ZipEntry(entry.getName());
-						newEntry.setTime(CONSTANT_TIME_FOR_ZIP_ENTRIES);
-						newEntry.setLastModifiedTime(FileTime.fromMillis(CONSTANT_TIME_FOR_ZIP_ENTRIES));
-						newEntry.setLastAccessTime(FileTime.fromMillis(CONSTANT_TIME_FOR_ZIP_ENTRIES));
+						setConstantFileTime(newEntry);
 					}
 
-					zipOutputStream.putNextEntry(newEntry);
-					InputStream inputStream = zipFile.getInputStream(entry);
-					byte[] buf = new byte[1024];
-					int length;
+					copyZipEntry(zipOutputStream, newEntry, zipFile.getInputStream(entry));
+				}
+			}
 
-					while ((length = inputStream.read(buf)) > 0) {
-						zipOutputStream.write(buf, 0, length);
-					}
+			try (var fileOutputStream = new FileOutputStream(file)) {
+				outZip.writeTo(fileOutputStream);
+			}
+		}
+	}
+
+	public static void appendZipEntry(File file, String path, byte[] data) throws IOException {
```

Comment:
Might want to add a comment about how this differs from `ZipUtils.add`

Reply:
Done 👍